src/sync/list.rs

```diff
@@ -223,14 +249,14 @@ impl<'g, T: 'g, C: Container<T>> Iterator for Iter<'g, T, C> {
                             // Deferred drop of `T` is scheduled here.
                             // This is okay because `.delete()` can be called only if `T: 'static`.
                             let p = self.curr;
-                            self.guard.defer(move || C::finalize(p.as_raw()));
+                            self.guard.defer(move || C::finalize(p.deref()));
                         }
                         self.curr = succ;
                     }
                     Err(err) => {
                         // We lost the race to delete the entry by a concurrent iterator. Set
                         // `self.curr` to the updated pointer, and report that we are stalled.
-                        self.curr = err.new;
+                        self.curr = err.current;
```

Comment:
@jeehoonkang Shouldn't we start again from the head here?

Reply:
You're right, we should!